pymc/printing.py

```diff
@@ -295,3 +297,26 @@ def _default_repr_pretty(obj: TensorVariable | Model, p, cycle):
 except (ModuleNotFoundError, AttributeError):
     # no ipython shell
     pass
+
+
+def _format_underscore(variable: str) -> str:
```

Comment:
Does this work equivalently?
```
import re

def _format_underscore(variable: str) -> str:
    """
    Escapes all unescaped underscores in the variable name for LaTeX representation.
    """
    return re.sub(r'(?<!\\)_', r'\\_', variable)
```

Reply:
Oh, that looks a lot better. Let me test it out.

Reply:
Yup, that seems to work equivalently! I will replace the long function with this.